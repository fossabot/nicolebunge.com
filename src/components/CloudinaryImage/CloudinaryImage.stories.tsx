import CloudinaryImage from '.';

export default {
  component: CloudinaryImage,
};

export const Default = {
  args: {
    alt: 'Image alt text',
    src: 'path/to/image.jpg',
  },
};
