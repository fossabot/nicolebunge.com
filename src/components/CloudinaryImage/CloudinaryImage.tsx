/* eslint-disable jsx-a11y/alt-text */

import React, { FC } from 'react';

export interface CloudinaryImageProps {
  alt: string;
  height?: number;
  src: string;
  width?: number;
}

const CloudinaryImage: FC<CloudinaryImageProps> = (props) => {
  const { src, ...otherProps } = props;

  return (
    <img
      loading="lazy"
      src={`https://res.cloudinary.com/nicolebunge/image/upload/f_auto,q_auto/v1627923793/nicolebunge.com/${src}`}
      {...otherProps}
    />
  );
};

export default CloudinaryImage;
