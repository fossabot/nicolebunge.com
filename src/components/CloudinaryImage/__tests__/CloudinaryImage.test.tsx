/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import CloudinaryImage from '..';

describe('Cloudinary image component', () => {
  it('renders correctly', () => {
    const { container } = render(<CloudinaryImage alt="Image alt text" src="path/to/image.jpg" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
