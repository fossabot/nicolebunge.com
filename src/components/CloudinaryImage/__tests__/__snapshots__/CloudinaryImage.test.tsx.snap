// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cloudinary image component renders correctly 1`] = `
<img
  alt="Image alt text"
  loading="lazy"
  src="https://res.cloudinary.com/nicolebunge/image/upload/f_auto,q_auto/v1627923793/nicolebunge.com/path/to/image.jpg"
/>
`;
